src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
 
     this.groundPlaces[placeGpuid] = newGroundPlace;
   }
+
+  /**
+   * @description Allows you to filter the ground places according to criteria such as name, gpuid, etc.
+   * @param {AutoCompleteFilters[]|undefined} filters - The filters you want to be applied on the search.
+   * @returns {AutoComplete[]}
+   */
+  public filterGroundPlaces(filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesAutocomplete: AutoComplete[] = this.parseGroundPlaces();
```

Comment:
You shouldn't reparse EVERY AUTOCOMPLETE that's crazy !

Reply:
It will work at first be remember you have a 10Mo file !

Reply:
Yes it's true, in fact find`, `filter`, `map`, `reduce` methods are much more efficient to use when the goal is to search trough something. So I convert my HashMap into a manipulable table to help me more easily.

Reply:
For this array of ground places, I've two options:
- Create a new parameter called groundPlacesAutocomplete that is instanciate and update like the first one, but the difference will be one the structure.
- Or only use this structure and so don't use HashMap object like, prefering an array of ground places instead..

Reply:
I will only have to use these ground places there, when I want to make a research on it.

Reply:
Start simple : 
* do something not optimized.
* then when everything will be in prod we will optimize it. 

but don't try to do a bit of both.

Creating a new structure EACH autocomplete is worse.